### ISSUE TITLE
Do not texture parts behind (hidden) 

### ISSUE THREAD

**Issue by remmel** [marcofugaro/three-projected-material#17]
Hi, thanks for your great lib!

It actually project the image on all part of the mesh, even if that part is behind (from the point of view of the camera used to project).
Do you plan to add a "visibility check" to avoid projecting the image on not visible parts ?
I only use 1 mesh.

![image](https://user-images.githubusercontent.com/937836/107255279-6b8c4600-6a38-11eb-83af-38a08bcf41d3.png)

Nothing special is done, but here is the code : https://github.com/remmel/image-processing-js/blob/45959089fbd042f5d3e986b73339131e238339be/pose-viewer/PoseCylinder.js#L62

http://remy-mellet.com/image-processing-js/pose-viewer.html?datasetType=LUBOS&datasetFolder=https://raw.githubusercontent.com/remmel/rgbd-dataset/main/20201130_195000.dataset

Thanks!

**Comment by marcofugaro** [comment]
Hmm I see, I didn't do this check for simplicity, it's quite elaborate and is called [Depth testing](https://learnopengl.com/Advanced-OpenGL/Depth-testing). But I'll see if I can make it work!

**Comment by remmel** [comment]
Yes, that's would be great :) on my side, I don't know anything about that shader topic!
